test/sql/hnsw_select.sql

```diff
@@ -71,5 +71,36 @@ SELECT has_index_scan('EXPLAIN (SELECT id FROM test1 ORDER BY v <-> ''{1,4}'') U
 -- Validate CTEs work and still use index
 SELECT has_index_scan('EXPLAIN WITH t AS (SELECT id FROM test1 ORDER BY v <-> ''{1,4}'') SELECT id FROM t UNION SELECT id FROM t');
 
+-- Validate <-> is replaced with the matching function when an index is present
+set enable_seqscan = true;
+EXPLAIN (COSTS false) SELECT * from small_world ORDER BY v <-> '{1,1,1}';
+SELECT * from small_world ORDER BY v <-> '{1,1,1}';
+begin;
+INSERT INTO test2 (v) VALUES ('{1,4}');
+INSERT INTO test2 (v) VALUES ('{2,4}');
+CREATE INDEX test2_cos ON test2 USING hnsw(v dist_cos_ops);
+EXPLAIN (COSTS false) SELECT * from test2 ORDER BY v <-> '{1,4}';
+-- Some additional cases that trigger operator rewriting
+-- SampleScan
+EXPLAIN (COSTS false) SELECT * FROM small_world TABLESAMPLE BERNOULLI (20) ORDER BY v <-> '{1,1,1}' ASC;
+-- SetOpt/HashSetOp
+EXPLAIN (COSTS false) (SELECT * FROM small_world ORDER BY v<-> '{1,0,1}' ASC )EXCEPT (SELECT * FROM small_world ORDER by v <-> '{1,1,1}' ASC LIMIT 5);
```

Comment:
Can you add another kind of test for these (described below)?

In sequential scans the use of `<->` operator should be identical to corresponding `_dist` function.
So, we should be able to run both and have a helper function that compares the output line by line and ensure they are the same.

Can you add this helper function and use it for these tests?